File: sif-guard/app/api/routes/health.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from sqlalchemy import text
from app.core.config import settings
from app.db.database import get_db
from app.services.sif.classifier import sif_classifier

router = APIRouter()


@router.get("/health")
def health_check(db: Session = Depends(get_db)):
    db_status = "healthy"
    try:
        db.execute(text("SELECT 1"))
    except Exception:
        db_status = "unhealthy"

    sif_model_status = "loaded" if sif_classifier.is_loaded else f"fallback_{sif_classifier.mode}"

    return {
        "status": "healthy" if (db_status == "healthy" and sif_classifier.is_loaded) else "degraded",
        "service": settings.PROJECT_NAME,
        "environment": settings.ENVIRONMENT,
        "components": {
            "database": db_status,
            "sif_model": sif_model_status,
            "sif_model_type": sif_classifier.mode,
            "embedding_model": settings.EMBEDDING_MODEL,
            "ocr_provider": settings.OCR_PROVIDER
        }
    }

File: sif-guard-frontend/src/api/health.ts
import { apiClient } from './client';

export interface HealthStatusResponse {
  status: string;
  service: string;
  environment: string;
  embedding_model?: string;
  components?: {
    database?: string;
    sif_model?: string;
    sif_model_type?: string;
    embedding_model?: string;
    ocr_provider?: string;
    [key: string]: string | undefined;
  };
  xgboost_model_loaded?: boolean;
  ocr_available?: boolean;
  bge_available?: boolean;
  lsr_available?: boolean;
  hdbscan_available?: boolean;
}

export async function checkSystemHealth(): Promise<HealthStatusResponse> {
  const res = await apiClient.get<HealthStatusResponse>('/health');
  return res.data;
}
